### PR TITLE
bulk API documentation Update

### PR DESCRIPTION
**Pull request by msvinaykumar**
## Description

Specified Bulk Service Configuration in bulkAPI MD file

    Datasource
    Limits
    Experiment Name
    Bulk API Threads

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

File: design/BulkAPI.md [hunk 1]
```diff
@@ -252,3 +252,61 @@ resource optimization in Kubernetes environments. Below is a breakdown of the JS
...
+        "url": "",
+        "authentication": {
+          "type": "bearer",
+          "credentials": {
+            "tokenFilePath": "/var/run/secrets/kubernetes.io/serviceaccount/token"
+          }
+        }
+      }
+    ]
+  }
```
Comment (dinogun) [inline design/BulkAPI.md:298]: Please mention that this is part of Kruize "initialization" and provide a link to the `openshift` yaml with a pointer to look for `kruizeconfigjson`

Comment (msvinaykumar) [reply design/BulkAPI.md:300]: done

File: design/BulkAPI.md [hunk 2]
```diff
@@ -252,3 +252,63 @@ resource optimization in Kubernetes environments. Below is a breakdown of the JS
...
+
+- **Default Limit:** Currently, the Bulk service supports only **1000 experiments** by default.
+- **Increasing the Limit:** You can increase this limit by setting the environment variable `bulkapilimit`.
+- **Job Failure on Exceeding Limit:** If the number of experiments exceeds the set limit, the job will fail.
+
+## Bulk API Threads
+
+- **Control Mechanism:** The number of threads used for bulk API operations can be controlled using the environment
+  variable `bulkThreadPoolSize`.
+- **With/Without KEDA Plugin:** This can be configured with or without the KEDA plugin. Ideally, the value should be
```
Comment (dinogun) [inline design/BulkAPI.md:313]: Sorry what is the KEDA plugin?

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**